mlflow/models/evaluation/default_evaluator.py

```diff
@@ -506,7 +491,8 @@ def _log_model_explainability(self):
             )
             return
 
-        if not (np.issubdtype(self.y.dtype, np.number) or self.y.dtype == np.bool_):
+        labels_dtype = self.y.copy_in_case_of_mutation().dtype
```

Comment:
Do we need to make a copy every single time `self.X` or `self.y` is accessed? It feels excessive and not as efficient.

Reply:
For current usages of `self.X`, we need a new copy every time with the exception of the `feature_dtypes` check in https://github.com/mlflow/mlflow/pull/6141/files#diff-1b3ab8e1152a42ab0bd3d4e01c7621b37a95aa02fff19642a5abf23f75b6e688R511. I'd rather err on the side of caution for `self.X` because we've seen multiple issues here.

There's no evidence that `self.y` needs to be copied, so I've gone ahead and reverted that overly-conservative change.